Update ViewCommand.java

minhao23 opened the pull request:
View Command is currently does not close automatically, let's update it such that windows are automatically closed at the moment when users open a new one.

minhao23 committed:
Update ViewCommand.java

View Command is currently does not close automatically, let's update it such that windows are automatically closed at the moment when users open a new one.

minhao23 committed:
Updated for checkstyle

starbucksventi commented on src/main/java/seedu/address/logic/commands/ViewCommand.java:
```diff
@@ -16,26 +16,19 @@
      * Constructs a {@code ViewCommand} with the specified {@code Index}.
      *
      * @param index The index of the person to view.
-     * @throws NullPointerException if the index is null.
```
Might want to add back that it throws a null pointer exception

starbucksventi reviewed (commented):
Lgtm after one nit👍

minhao23 committed:
Merge branch 'master' into backgound-colour

minhao23 committed:
Add close function to view

minhao23 committed:
Add new UI updates

This current commit will bring the UI up to date with previous discussions, and implement the Single Page View with Close command.

Tests have also been implemented.

starbucksventi commented on src/main/java/seedu/address/logic/commands/ViewCommand.java:
```diff
@@ -89,11 +95,22 @@ public CommandResult execute(Model model) throws CommandException {
+        }
+    }
+
+    public static Stage getCurrentStage() {
```
Might need to add JavaDoc for this

starbucksventi commented on src/main/java/seedu/address/logic/parser/exceptions/NoWindowException.java:
```diff
@@ -0,0 +1,7 @@
+package seedu.address.logic.parser.exceptions;
+
+public class NoWindowException extends RuntimeException{
```
Add JavaDoc comment

starbucksventi commented on src/main/java/seedu/address/ui/PersonCard.java:
```diff
@@ -70,11 +66,11 @@ public PersonCard(Person person, int displayedIndex) {
-            Label historyLabel = new Label(historyText.toString());
-            history.getChildren().add(historyLabel);
-        });
+//        person.getHistory().getHistoryEntries().forEach((date, activities) -> {
```
Might want to remove the commented portion

starbucksventi commented on src/test/java/seedu/address/logic/commands/CloseCommandTest.java:
```diff
@@ -0,0 +1,50 @@
+    public void closeCurrentWindow_noWindowOpen_throwsNoWindowException() {
+        assertThrows(NoWindowException.class, ViewCommand::closeCurrentWindow, ViewCommand.NO_WINDOWS_OPEN);
+    }
+//    @Test
```
Delete commented chunk

minhao23 committed:
Update commit for checkstyle

minhao23 committed:
Update commit for checkstyle
